controller/linodecluster_controller.go

```diff
@@ -180,6 +181,18 @@ func (r *LinodeClusterReconciler) reconcile(
 		conditions.MarkTrue(clusterScope.LinodeCluster, ConditionPreflightLinodeVPCReady)
 	}
 
+	// Check if the firewall is created
+	if clusterScope.LinodeCluster.Spec.NodeBalancerFirewallRef != nil {
+		if !reconciler.ConditionTrue(clusterScope.LinodeCluster, ConditionPreflightLinodeFirewallReady) {
+			res, err := r.reconcilePreflightLinodeFirewallCheck(ctx, logger, clusterScope)
+			if err != nil || !res.IsZero() {
+				conditions.MarkFalse(clusterScope.LinodeCluster, ConditionPreflightLinodeFirewallReady, string("linode firewall not yet available"), clusterv1.ConditionSeverityError, "")
+				return res, err
+			}
+		}
+		conditions.MarkTrue(clusterScope.LinodeCluster, ConditionPreflightLinodeFirewallReady)
```

Comment:
Should we change this to `ConditionPreflightNodebalancerFirewallReady`?

Reply:
Hmmm good question! Makes sense to switch it since we are dealing with different firewalls here. Let me update that

Reply:
Done!